x/evm/keeper/state_transition.go

```diff
@@ -141,29 +141,22 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 	logs := k.GetTxLogs(txHash)
 
 	// Commit and switch to original context
-	if !res.Reverted {
+	if len(res.VmError) == 0 {
```

Comment:
improvement for later, create `Reverted()` util function for `MsgEthereumTxResponse`

Reply:
fixed, called `Failed`.